llvm fails to build without full Xcode

llvm fails to build from source without full Xcode. This was introduced with #34419 (commit 38c4a8e43d2) where toolchain build was forced and `--with-toolchain` option removed.

I have to build LLVM from source because I need new _lldb_ which is optional (`--with-lldb`) in the formula however I do not want to install full Xcode.

Build output snippet:
```
==> cmake -G Unix Makefiles .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Ce
Last 15 lines from /Users/jvcelak/Library/Logs/Homebrew/llvm/02.cmake:
  command.


CMake Warning at tools/xcode-toolchain/CMakeLists.txt:60 (message):
  Failed to detect the version of an installed copy of Xcode, falling back to
  highest supported version.  Set XCODE_VERSION to override.


CMake Error at tools/xcode-toolchain/CMakeLists.txt:80 (message):
  Could not identify toolchain dir


-- Configuring incomplete, errors occurred!
```

This is what fails in llvm's cmake:
```
$ xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

I will be happy to open a PR with the fix but would like to consult the solution first. Here are some options:

1. Add option `--without-toolchain` to allow disabling of Xcode toolchain build. Users will have the toolchain in bottles but there will be a way to rebuild llvm without full Xcode.
2. Always build _lldb_ and remove `--with-lldb` option. This is not general solution but eliminates my need of building LLVM from source. Problem is that _lldb_ needs to be code-signed to work properly. Users would have to sign lldb after installation manually.
3. Revert #34419 entirely: No toolchain in bottle and `--with-toolchain` option.

Other suggestions are welcome as well. What do you prefer?

---

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

--- a/Formula/llvm.rb
+++ b/Formula/llvm.rb
@@ -111,6 +111,7 @@ class Llvm < Formula
 
   # https://llvm.org/docs/GettingStarted.html#requirement
   depends_on "cmake" => :build
+  depends_on :xcode => :build
   depends_on "libffi"
 
   depends_on "python@2" if MacOS.version <= :snow_leopard

